frontends/concrete-python/examples/sha1/sha1.md

```diff
@@ -0,0 +1,307 @@
+# SHA1 Computation with Modules
+
+We take the example of a SHA1 computation to show how
+[Modules](https://docs.zama.ai/concrete/compilation/modules) can be useful in Concrete. SHA1 is a
+deprecated broken hash function: we didn't take this example for its security, but rather as it's a
+pedagogical example of how Modules can be useful.
+
+The SHA1 code is given [here](sha1.py). Execution times of the different functions are given in the
+final section.
+
+We made our example by forking
+[https://github.com/ajalt/python-sha1](https://github.com/ajalt/python-sha1) by
+[AJ Alt](https://github.com/ajalt), and then made lot of modifications to have an FHE implementation
+and corresponding tests.
+
+## SHA1 Overview
+
+SHA1 is a deprecated broken hash function defined by NIST in 1995: one can find information about it
+in its [Wikipedia page](https://en.wikipedia.org/wiki/SHA-1), or in its
+[official description](https://nvlpubs.nist.gov/nistpubs/fips/nist.fips.180-4.pdf).
+
+Its [pseudo-code](https://en.wikipedia.org/wiki/SHA-1#SHA-1_pseudocode) is particularly interesting,
+since our implementation in Concrete Python will follow the structure.
```

Comment:
> one can find information about it
> in its [Wikipedia page](https://en.wikipedia.org/wiki/SHA-1), or in its
> [official description](https://nvlpubs.nist.gov/nistpubs/fips/nist.fips.180-4.pdf).
> 
> Its [pseudo-code](https://en.wikipedia.org/wiki/SHA-1#SHA-1_pseudocode) is particularly interesting,
> since our implementation in Concrete Python will follow the structure.

**Change to:**
You can find detailed information on SHA1 on its [Wikipedia page](https://en.wikipedia.org/wiki/SHA-1) or in its [official description](https://nvlpubs.nist.gov/nistpubs/fips/nist.fips.180-4.pdf). We follow the structure of its [pseudo-code](https://en.wikipedia.org/wiki/SHA-1#SHA-1_pseudocode)  in our implementation.

Reply:
done